src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -5748,136 +5754,144 @@ void func_808355D8(PlayState* play, Player* this, PlayerAnimationHeader* anim) {
-                return true;
             }
-        } else {
-            if ((this->stateFlags1 & PLAYER_STATE1_8000000) && (this->actor.floorPoly == NULL)) {
-                BgCheck_EntityRaycastFloor7(&play->colCtx, &this->actor.floorPoly, &sp30, &this->actor,
-                                            &this->actor.world.pos);
-                if (this->actor.floorPoly == NULL) {
-                    func_80169EFC(&play->state);
-                    return false;
+
+            if (!(this->stateFlags1 & (PLAYER_STATE1_800000 | PLAYER_STATE1_8000000 | PLAYER_STATE1_20000000)) &&
+                ((floorType = SurfaceType_GetFloorType(&play->colCtx, poly, bgId)) != FLOOR_TYPE_10) &&
+                ((sp34 < 100) || (this->actor.bgCheckFlags & BGCHECKFLAG_GROUND))) {
+                if (floorType == FLOOR_TYPE_11) {
+                    func_8019F128(NA_SE_OC_SECRET_HOLE_OUT);
+                    func_801A4058(5);
+                    gSaveContext.seqId = (u8)NA_BGM_DISABLED;
+                    gSaveContext.ambienceId = AMBIENCE_ID_DISABLED;
+                } else {
+                    func_8085B74C(play);
                 }
-                //! FAKE
-                if (1) {}
+            } else if (!(this->actor.bgCheckFlags & BGCHECKFLAG_GROUND)) {
+                Player_StopHorizontalMovement(this);
             }
 
-            if (!(this->stateFlags1 & PLAYER_STATE1_80000000)) {
-                if (((this->actor.world.pos.y < -4000.0f) ||
-                     (((this->floorProperty == FLOOR_PROPERTY_5) || (this->floorProperty == FLOOR_PROPERTY_12) ||
-                       (this->floorProperty == FLOOR_PROPERTY_13)) &&
-                      ((D_80862B18 < 100.0f) || (this->fallDistance > 400))))) {
-                    if (this->actor.bgCheckFlags & BGCHECKFLAG_GROUND) {
-                        if (this->floorProperty == FLOOR_PROPERTY_5) {
-                            func_80169FDC(&play->state);
-                            func_808345C8();
-                        } else {
-                            func_80169EFC(&play->state);
-                        }
-                        if (!SurfaceType_IsWallDamage(&play->colCtx, this->actor.floorPoly, this->actor.floorBgId)) {
-                            gSaveContext.respawnFlag = -5;
-                        }
+            Camera_ChangeSetting(Play_GetCamera(play, CAM_ID_MAIN), CAM_SET_SCENE0);
+            this->stateFlags1 |= PLAYER_STATE1_1 | PLAYER_STATE1_20000000;
+            return true;
+        }
+        if ((this->stateFlags1 & PLAYER_STATE1_8000000) && (this->actor.floorPoly == NULL)) {
+            BgCheck_EntityRaycastFloor7(&play->colCtx, &this->actor.floorPoly, &sp30, &this->actor,
+                                        &this->actor.world.pos);
+            if (this->actor.floorPoly == NULL) {
+                func_80169EFC(&play->state);
+                return false;
+            }
+            //! FAKE
+            if (1) {}
```

Comment:
Confused here, could just be how the diff is, but does this add a fake, or just move the fake from line 5831 to here?

Reply:
From the looks of it, kind of move, but it isn't that the fake moved (it is still in the same block of code), something changed so it looks like whole blocks of code got moved.

Reply:
Yeah, there was an `} else {` right after a `return false` which added an extra indentation for no reason. Unfortunately, GitHub isn't great at displaying diffs when the indentation changes for large chunks, but the fake `if (1) {}` was there before